### PR TITLE
Refactor/remove powermockito

### PR DESCRIPTION
**Pull request by frisonisland**
This PR will remove Powermockito.
Changes have been cherry-picked from [feat/copy-to-drive-api](https://github.com/Zextras/carbonio-mailbox/tree/feat/copy-to-drive-api)

refactor:
- add Mockito library to mock static
- refactor many test classes to remove PowerMockito
- some non-test classes have an added method to enable easier mocking, instead of calling a static class method directly

### CODE REVIEW

**Review by keshavbhatt — APPROVED** [review]

File: store/src/java-test/com/zimbra/cs/imap/MemcachedImapCacheTest.java [hunk 1]
```diff
@@ -56,10 +51,10 @@ public void tearDown() throws Exception {
     public void testInvalidObject() {
         try {
 
-            PowerMockito.mockStatic(MemcachedConnector.class);
+            Mockito.mockStatic(MemcachedConnector.class);
```
Comment (Polpetta) [inline store/src/java-test/com/zimbra/cs/imap/MemcachedImapCacheTest.java:54]: Use static import

File: store/src/java-test/com/zimbra/cs/mailbox/ContactTest.java [hunk 2]
```diff
@@ -323,17 +320,31 @@ public void testEncodeContact() throws Exception {
...
-        Assert.assertFalse(ToXML.hasDLViewRight("mydl@zimbra.com", account, account));
+        try (MockedStatic<ToXML> toXMLMockedStatic = Mockito.mockStatic(ToXML.class)) {
+            GalGroupInfoProvider galGroupInfoProvider = Mockito.mock(GalGroupInfoProvider.class);
+            Mockito.doReturn(GroupInfo.IS_GROUP)
+                .when(galGroupInfoProvider).getGroupInfo(Mockito.anyString(), Mockito.anyBoolean(),
+                    Mockito.any(Account.class), Mockito.any(Account.class));
+            toXMLMockedStatic.when(ToXML::getGalGroupInfoProvider).thenReturn(galGroupInfoProvider);
+            toXMLMockedStatic.when(() -> ToXML.hasDLViewRight("mydl@zimbra.com", account, account))
+                .thenCallRealMethod();
+            Assert.assertFalse(ToXML.hasDLViewRight("mydl@zimbra.com", account, account));
```
Comment (Polpetta) [inline store/src/java-test/com/zimbra/cs/mailbox/ContactTest.java:331]: Use static import

File: store/src/java-test/com/zimbra/cs/mailbox/ContactTest.java [hunk 3]
```diff
@@ -323,17 +320,31 @@ public void testEncodeContact() throws Exception {
...
+        GalGroupInfoProvider galGroupInfoProvider = Mockito.mock(GalGroupInfoProvider.class);
+        // inside try logic to avoid Mockito exception when mocking static same class (Thread scope)
+        try (MockedStatic<ToXML> toXMLMockedStatic = Mockito.mockStatic(ToXML.class)) {
+            Mockito.doReturn(GroupInfo.IS_GROUP)
+                .when(galGroupInfoProvider).getGroupInfo(Mockito.anyString(), Mockito.anyBoolean(),
+                    Mockito.any(Account.class), Mockito.any(Account.class));
+            toXMLMockedStatic.when(ToXML::getGalGroupInfoProvider).thenReturn(galGroupInfoProvider);
+            toXMLMockedStatic.when(() -> ToXML.hasDLViewRight(null, account, account)).thenCallRealMethod();
+            Assert.assertTrue(ToXML.hasDLViewRight(null, account, account));
+        }
```
Comment (Polpetta) [inline store/src/java-test/com/zimbra/cs/mailbox/ContactTest.java:347]: Use static import

File: store/src/java-test/com/zimbra/cs/service/UserServletTest.java [hunk 4]
```diff
@@ -1,407 +1,389 @@
...
+  public void testDoGet() {
+    HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+    MockHttpServletResponse response = new MockHttpServletResponse();
+    UserServlet userServlet = new UserServlet();
+    try {
+
+      spy(ZimbraServlet.class);
+      spy(UserServlet.class);
+
+      Mockito.mockStatic(L10nUtil.class);
```
Comment (Polpetta) [inline store/src/java-test/com/zimbra/cs/service/UserServletTest.java:57]: Use static import

File: store/src/java-test/com/zimbra/cs/service/mail/SearchActionTest.java [hunk 5]
```diff
@@ -139,8 +134,13 @@ public void testSearchActionRead() throws Exception {
...
         ConvAction convAction = new ConvAction();
-        PowerMockito.stub(PowerMockito.method(SoapHttpTransport.class, "invokeWithoutSession"))
-            .toReturn(
+        SoapHttpTransport mockSoapHttpTransport = Mockito.mock(SoapHttpTransport.class);
+        MockedStatic<SearchAction> searchActionMockedStatic = Mockito.mockStatic(SearchAction.class);
+        searchActionMockedStatic.when(() -> SearchAction.getSoapHttpTransportInstance(Mockito.anyString()))
+                .thenReturn(mockSoapHttpTransport);
+
+        Mockito.when(mockSoapHttpTransport.invokeWithoutSession(Mockito.any()))
+                .thenReturn(
```
Comment (Polpetta) [inline store/src/java-test/com/zimbra/cs/service/mail/SearchActionTest.java:143]: Use static import

**Review by Polpetta — APPROVED** [review]
There are some static imports still in use, other than that it's a :+1: for me

**Comment by frisonisland** [comment]
> There are some static imports still in use, other than that it's a +1 for me

Thanks @Polpetta , I've updated with static import and marked as resolved. Now will merge!

**Pull request merged**